pkg/validate/openshift_cluster.go

```diff
@@ -0,0 +1,216 @@
+package validate
```

Comment:
@pweil- @kargakis @mjudeikis combined validation as discussed

Reply:
for the future, what if we put all error messages to something like:
```
//go:generate stringer -type=ErrorType
type ErrorType int

const (
	ErrorMalformedManifest ErrorType = iota
	ErrorInvalidOpenshiftVersion
)
```
This would enable easier testing when `[]error` returned.

Reply:
MSFT have specific requirements around errors which they haven't shared with us yet (@amanohar please can you do this?) - we'll need to take these into account.